server/coordinator/procedure/storage.go

```diff
@@ -18,6 +18,13 @@ type Meta struct {
 	RawData []byte
 }
 
+func (m *Meta) needRetry() bool {
+	if m.State == StateCancelled || m.State == StateFinished {
+		return false
+	}
+	return true
+}
```

Comment:
`needRetry` should move to schedule?

Reply:
`Scheduler` will be added in next pull request, now I just record todo here and resolve this conversation.